code/modules/mob/living/simple_animal/hostile/goose.dm

```diff
@@ -25,33 +25,76 @@
 	faction = list("neutral")
 	attack_same = TRUE
 	gold_core_spawnable = HOSTILE_SPAWN
-	var/icon_resting = "goose_sit"
+	var/random_retaliate = TRUE
+	var/icon_vomit_start = "vomit_start"
+	var/icon_vomit = "vomit"
+	var/icon_vomit_end = "vomit_end"
 
-
-/mob/living/simple_animal/hostile/retaliate/goose/toggle_ai(togglestatus)
+/mob/living/simple_animal/hostile/retaliate/goose/handle_automated_movement()
 	. = ..()
-	if(!key)
-		if(AIStatus != AI_ON)
-			set_resting(TRUE)
-		else
-			set_resting(FALSE)
+	if(prob(5) && random_retaliate == TRUE)
+		Retaliate()
 
-/mob/living/simple_animal/hostile/retaliate/goose/update_resting()
+/mob/living/simple_animal/hostile/retaliate/goose/vomit //https://cdn.discordapp.com/attachments/429431032228610058/585549032177401857/vomitgoose.png
+	name = "Birdboat"
+	real_name = "Birdboat"
+	desc = "It's a sick-looking goose. Probably ate too much maintenance trash."
+	gender = MALE
+	response_help  = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm   = "kicks"
+	gold_core_spawnable = NO_SPAWN
+	random_retaliate = FALSE
+	var/vomiting = FALSE
+	var/datum/action/cooldown/vomit/goosevomit
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/Initialize()
 	. = ..()
-	if(resting)
-		wander = FALSE
-	else
-		wander = TRUE
-	update_icon()
+	goosevomit = new()
+	goosevomit.Grant(src)
 
-/mob/living/simple_animal/hostile/retaliate/goose/handle_automated_movement()
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/Destroy()
+	QDEL_NULL(goosevomit)
+	return ..()
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/vomit()
+	var/turf/T = get_turf(src)
+	playsound(get_turf(src), 'sound/effects/splat.ogg', 50, 1) //yes getting the turf twice is necessary fuck off
+	T.add_vomit_floor(src)
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/vomit_start(duration)
+	flick("vomit_start",src)
+	vomiting = TRUE
+	icon_state = "vomit"
+	vomit()
+	addtimer(CALLBACK(src, .proc/vomit_end), duration)
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/vomit_end()
+	flick("vomit_end",src)
+	vomiting = FALSE
+	icon_state = initial(icon_state)
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/Moved(oldLoc, dir)
 	. = ..()
-	if(prob(5))
-		Retaliate()
+	if(vomiting)
+		vomit() // its supposed to keep vomiting if you move
+		return
+	if(prob(0.5))
```

Comment:
use `if(!vomiting && prob(0.05))` so that it doesn't unfairly get to roll to start vomiting while already vomiting

Reply:
the animation only plays when called by vomit_start, i have to keep the if statements seperate so it doesnt only play the animation once you stop moving

Reply:
that wouldnt do anything because the if(vomiting) before it already returns

Reply:
shit you right

Reply:
oh yeah, it's actually fine never mind then